### PR TITLE
Refactor : src/components/AgendaItems/AgendaItemsCreateModal from Jest to Vitest #2489 

### PR DESCRIPTION
**Pull request by Ramneet04**
Refactored src/components/AgendaItems/AgendaItemsCreateModal.test.tsx to src/components/AgendaItems/AgendaItemsCreateModal.spec.tsx

Issue: https://github.com/PalisadoesFoundation/talawa-admin/issues/2489 : https://github.com/PalisadoesFoundation/talawa-admin/issues/2489

Changes Implemented:

Refactored the testing framework from Jest to Vitest in AgendaItemsCreateModal.test.tsx.
Updated import statements, mocking methods, and assertions to align with Vitest conventions.
Verified compatibility with the existing codebase using Vitest.
- Renamed the test file:
From AgendaItemsCreateModal.test.tsx → AgendaItemsCreateModal.spec.tsx to follow the naming convention for Vitest.
- Ran all tests successfully under the Vitest environment.
https://github.com/PalisadoesFoundation/talawa-admin/issues/2489

migrateIssue

Other information
I have read the previous refactor PR and tried to keep things as uniform as possible.
Please suggest any other changes if required.

## Summary by CodeRabbit

- **Tests**
  - Transitioned testing framework from Jest to Vitest for improved mocking and testing utilities.
  - Enhanced testing capabilities with updated mock function declarations and import statements.
  - Maintained existing test structure and logic, ensuring consistent functionality and behavior verification for the `AgendaItemsCreateModal` component.

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@Ramneet04 Please fix the first comment so that each issue listed automatically closes. The PR_GUIDELINES.md file has details.

Please address the coderabbit comments and fix the failed tests.

**Comment by palisadoes** [comment]
Please fix the failing test and ensure that CodeRabbit.ai approves your PR

**Comment by Ramneet04** [comment]
hey @palisadoes  could you please assist me like why this one failed, I have correct the technical error but don't know this one.
@codecov
codecov/project — 87.08% (-5.16%) compared to debe672

**Comment by palisadoes** [comment]
Please ensure that coderabbit.ai approves your changes. The error may be corrected after that.

**Comment by Ramneet04** [comment]
Finally!! Is there anything or is it resolved??

File: src/components/AgendaItems/AgendaItemsCreateModal.spec.tsx [hunk 1]
```diff
@@ -19,6 +19,10 @@ import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
 import AgendaItemsCreateModal from './AgendaItemsCreateModal';
 import { toast } from 'react-toastify';
 import convertToBase64 from 'utils/convertToBase64';
+import type { MockedFunction } from 'vitest';
+
+import '@testing-library/jest-dom';
```
Comment (palisadoes) [inline src/components/AgendaItems/AgendaItemsCreateModal.spec.tsx:24]: Why was this added if we are migrating from Jest?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: src/components/AgendaItems/AgendaItemsCreateModal.spec.tsx (hunk 1 above)
Comment (Ramneet04) [reply src/components/AgendaItems/AgendaItemsCreateModal.spec.tsx]: I was going though the guide I though it is necessary but its not.
I have added screenshot as well it works after removal
lastChange

**Review by palisadoes — APPROVED** [review]

**Pull request merged**